plugins/module_utils/openshift_adm_prune_images.py

```diff
@@ -0,0 +1,514 @@
+                ).to_dict()
+            except DynamicApiError as exc:
+                msg = "Failed to patch object: kind={0} {1}/{2}".format(
+                    kind, namespace, name
+                )
+                self.fail_json(msg=msg, status=exc.status, reason=exc.reason)
+            except Exception as exc:
+                msg = "Failed to patch object kind={0} {1}/{2} due to: {3}".format(
+                    kind, namespace, name, exc
+                )
+                self.fail_json(msg=msg, error=to_native(exc))
+        return result
+
+    def delete_image(self, image):
+        kind = "Image"
+        api_version = "image.openshift.io/v1"
+        resource = self.find_resource(kind=kind, api_version=api_version)
+        name = image["metadata"]["name"]
+        self.changed = True
+        if not self.check_mode:
+            try:
+                delete_options = client.V1DeleteOptions(grace_period_seconds=0)
+                return resource.delete(name=name, body=delete_options).to_dict()
+            except NotFoundError:
+                pass
+            except DynamicApiError as exc:
+                self.fail_json(
+                    msg="Failed to delete object %s/%s due to: %s" % (
+                        kind, name, exc.body
+                    ),
+                    reason=exc.reason,
+                    status=exc.status
+                )
+        else:
+            existing = resource.get(name=name)
+            if existing:
+                existing = existing.to_dict()
+            return existing
+
+    def exceeds_limits(self, namespace, image):
+        if namespace not in self.limit_range:
+            return False
+        docker_image_metadata = image.get("dockerImageMetadata")
+        if not docker_image_metadata:
+            return False
+        docker_image_size = docker_image_metadata["Size"]
+
+        for limit in self.limit_range.get(namespace):
+            for item in limit["spec"]["limits"]:
+                if item["type"] != "openshift.io/Image":
+                    continue
+                limit_max = item["max"]
+                if not limit_max:
+                    continue
+                storage = limit_max["storage"]
+                if not storage:
+                    continue
+                if convert_storage_to_bytes(storage) < docker_image_size:
+                    # image size is larger than the permitted limit range max size
+                    return True
+        return False
+
+    def prune_image_stream_tag(self, stream, tag_event_list):
+        manifests_to_delete, images_to_delete = [], []
+        filtered_items = []
+        tag_event_items = tag_event_list["items"] or []
+        prune_over_size_limit = self.params.get("prune_over_size_limit")
+        stream_namespace = stream["metadata"]["namespace"]
+        stream_name = stream["metadata"]["name"]
+        for idx in range(len(tag_event_items)):
+            item = tag_event_items[idx]
```

Comment:
This appears in a few methods here, and while correct, it's a bit awkward to read. This is just a suggestion, but the following would probably be more readable:
```suggestion
        for idx, item in enumerate(tag_event_items):
```

Reply:
will consider it 
Thanks